testsuite/test_wbemconnection_mock.py

```diff
@@ -4332,3 +4394,176 @@ def test_invokemethod(self, conn, ns, desc, inputs, exp_output,
+        assert schema.schema_version_str == '2.49.0'
+        assert os.path.isdir(schema.schema_root_dir)
+        assert os.path.isdir(schema.schema_mof_dir)
+        assert os.path.isfile(schema.schema_mof_filename)
+        assert schema.schema_root_dir == test_schema_root_dir
+        mof_dir = 'mofFinal%s' % schema.schema_version_str
+        test_schema_mof_dir = os.path.join(test_schema_root_dir, mof_dir)
+        assert schema.schema_mof_dir == test_schema_mof_dir
+
+        assert schema.schema_mof_filename == \
+            os.path.join(test_schema_mof_dir,
+                         'cim_schema_%s.mof' % (schema.schema_version_str))
+
+        assert repr(schema) == 'DMTFCIMSchema(' \
+                               'schema_version=%s, ' \
+                               'schema_root_dir=%s, schema_mof_dir=%s, ' \
+                               'schema_mof_filename=%s)' % \
+                               (DMTF_TEST_SCHEMA_VER, test_schema_root_dir,
+                                schema.schema_mof_dir,
+                                schema.schema_mof_filename)
+
+        schema_mof = schema.build_schema_mof(['CIM_ComputerSystem', 'CIM_Door'])
+
+        exp_mof = '#pragma locale ("en_US")\n' \
+                  '#pragma include ("System/CIM_ComputerSystem.mof")\n' \
+                  '#pragma include ("Device/CIM_Door.mof")\n'
+
+        assert schema_mof == exp_mof
+
+    def test_schema_experimental_load(self, test_schema_root_dir):
+        # pylint: disable=no-self-use
+        """
+        Test the DMTFCIMSchema class and its methods to get a schema from the
+        DMTF, expand it, and to create a partial MOF schema definition for
+        compilation.
+        """
+        schema = DMTFCIMSchema(DMTF_TEST_SCHEMA_VER, test_schema_root_dir,
+                               use_experimental=True,
+                               verbose=True)
+
+        assert schema.schema_version == DMTF_TEST_SCHEMA_VER
+        assert schema.schema_version_str == '2.49.0'
+        assert os.path.isdir(schema.schema_root_dir)
+        assert os.path.isdir(schema.schema_mof_dir)
+        assert os.path.isfile(schema.schema_mof_filename)
+        assert schema.schema_root_dir == test_schema_root_dir
+        mof_dir = 'mofExperimental%s' % schema.schema_version_str
+        test_schema_mof_dir = os.path.join(test_schema_root_dir, mof_dir)
+        assert schema.schema_mof_dir == test_schema_mof_dir
+
+        assert schema.schema_mof_filename == \
+            os.path.join(test_schema_mof_dir,
+                         'cim_schema_%s.mof' % (schema.schema_version_str))
+
+    def test_two_schema_load(self, test_schema_root_dir):
+        # pylint: disable=no-self-use
+        """Test loading two different schema"""
+        schema1 = DMTFCIMSchema(DMTF_TEST_SCHEMA_VER, test_schema_root_dir,
+                                verbose=True)
+
+        schema2 = DMTFCIMSchema((2, 50, 0), test_schema_root_dir, verbose=True)
```

Comment:
This downloads from the DMTF web site, because that version is not persisted. I read your reply and agree with the motivation to test that.

My suggestion is to control this additional test via an environment variable (e.g. TEST_SCHEMA_DOWNLOAD=m.n.u) , which we would document in the help text of the makefile.

Reply:
Good call.  Redid those tests.  This could be reduced to a single method with parametarization I think but did not want to do that today.

DONE